src/PortfolioPlanning/Components/Plan/AddItemPanel.tsx

```diff
@@ -272,26 +276,163 @@ export class AddItemPanel extends React.Component<IAddItemPanelProps, IAddItemPa
                         No work items of this type were found in the project.
                     </div>
                 );
-            } else if (content.items.length === 0) {
-                return <div className="workItemTypeEmptyMessage">All work items are already added to plan.</div>;
+            } else if (Object.keys(content.items).length === 0) {
+                return <div className="workItemTypeEmptyMessage">All work items were already added to the plan.</div>;
             } else {
                 if (!this._selectionByWorkItemType[workItemTypeKey]) {
-                    this._selectionByWorkItemType[workItemTypeKey] = new ListSelection(true);
+                    const selection = new Selection({
+                        onSelectionChanged: () => this._onWorkItemSelectionChanged(workItemTypeKey)
+                    });
+
+                    this._selectionByWorkItemType[workItemTypeKey] = selection;
                 }
 
-                return (
-                    <ScrollableList
-                        className="item-list"
-                        itemProvider={new ArrayItemProvider<IListBoxItem>(content.items)}
-                        renderRow={this.renderRow}
+                let allItemsCount: number = 0;
+                const listItems: IAddItem[] = [];
+                Object.keys(content.items).forEach(workItemId => {
+                    const item = content.items[workItemId];
+                    allItemsCount++;
+
+                    if (item.hide === false) {
+                        listItems.push(item);
+                    }
+                });
+
+                const columns: IColumn[] = [
```

Comment:
Is it worth putting the icon, id, and title all in separate columns?

Reply:
if we wanted to sort by column, maybe... not needed for now

Reply:
Makes sense. Good to know about.